roachprod-microbench,build: ignore packages

herkolategan opened the pull request:
Recently, `pkg/testutils/lint/lint_test.go` (https://github.com/cockroachdb/cockroach/pull/119589) added logic to `init`, that requires
a `GO_SDK` env var that causes a failure when listing the benchmarks for that
package, since on the target microbenchmark env there is no such env var
specified. It's possible to add the env var to avoid failure. But in the future
more logic that could probe the env var further could still cause a failure as
go will not be installed on these machines. These packages should be excluded
from any interaction, to avoid complete execution failure.

This is slightly different from the other `exclude` flag that is meant to not
execute specific benchmarks within a given package.

See: https://github.com/cockroachdb/cockroach/pull/119589

Epic: None
Release Note: None

herkolategan committed:
roachprod-microbench: ignore incompatible packages

Recently, `pkg/testutils/lint/lint_test.go` (#119589) added logic to `init`, that requires
a `GO_SDK` env var that causes a failure when listing the benchmarks for that
package, since on the target microbenchmark env there is no such env var
specified. It's possible to add the env var to avoid failure. But in the future
more logic that could probe the env var further could still cause a failure as
go will not be installed on these machines. These packages should be excluded
from any interaction, to avoid complete execution failure.

This is slightly different from the other `exclude` flag that is meant to not
execute specific benchmarks within a given package.

See: #119589

Epic: None
Release Note: None

cockroach-teamcity commented:
<!-- Reviewable:start -->
This change is [<img src="https://reviewable.io/review_button.svg" height="34" align="absmiddle" alt="Reviewable"/>](https://reviewable.io/reviews/cockroachdb/cockroach/120752)
<!-- Reviewable:end -->

renatolabs commented:
Where would this `--no-probe` flag be added? Do we not keep the scripts under source control?

srosenberg commented:
> Where would this `--no-probe` flag be added? Do we not keep the scripts under source control?

Methinks new flag needs to be integrated with `build/teamcity/cockroach/nightlies/microbenchmark_weekly.sh`.

srosenberg commented:
Maybe we could just avoid the `panic` in lint_test.go by setting `GO_SDK`? I suppose there could be other (future) changes which would need to be excluded. Btw, the (commit) description is a bit fuzzy; had to find the breaking change in the linked issue to see which `init` was panicking.

srosenberg commented on pkg/cmd/roachprod-microbench/main.go:
```diff
@@ -89,7 +89,8 @@ func makeRunCommand() *cobra.Command {
 	cmd.Flags().StringVar(&config.shellCommand, "shell", config.shellCommand, "additional shell command to run on node before benchmark execution")
-	cmd.Flags().StringSliceVar(&config.excludeList, "exclude", []string{}, "comma-separated regex of packages and benchmarks to exclude e.g. 'pkg/util/.*:BenchmarkIntPool,pkg/sql:.*'")
+	cmd.Flags().StringSliceVar(&config.excludeList, "exclude", []string{}, "benchmarks to exclude, in the form <pkg regex:benchmark regex> e.g. 'pkg/util/.*:BenchmarkIntPool,pkg/sql:.*'")
+	cmd.Flags().StringSliceVar(&config.noProbePkgList, "no-probe", []string{}, "packages to completely exclude from probing or execution'")
```
Nit: `no-probe` is kinda awkward; why not `ignore-packages`?

herkolategan replied:
Yep, naming was not my strong suite today. `ignore-packages` is clearer.

herkolategan commented:
> Maybe we could just avoid the `panic` in lint_test.go by setting `GO_SDK`? I suppose there could be other (future) changes which would need to be excluded. Btw, the (commit) description is a bit fuzzy; had to find the breaking change in the linked issue to see which `init` was panicking.

I'll update the commit to point out the specific package & failure, as to avoid having to drill into the linked issue.

herkolategan commented:
> > Where would this `--no-probe` flag be added? Do we not keep the scripts under source control?
> 
> Methinks new flag needs to be integrated with `build/teamcity/cockroach/nightlies/microbenchmark_weekly.sh`.

Yes, I'll add that as a commit to this PR. For some reason thought I would be able to pass it through TC directly, but it's not the case.

herkolategan committed:
build: microbenchmark ignore packages

Pass the `BENCH_IGNORE_PACKAGES` var to `roachprod-microbench`.

Epic: None
Release Note: None

herkolategan commented:
TFTRs!

bors r=renatolabs,srosenberg
